spec/linked_list.js

```diff
@@ -0,0 +1,205 @@
+import chai, { expect } from 'chai'
+import chaiChange from 'chai-change'
+import LinkedList from '../src/linked_list'
+
+chai.use(chaiChange)
+
+describe('Linked List', () => {
+  'use strict'
+
+  it('exists', () => {
+    expect(LinkedList).to.be.a('function')
+  })
+
+  context('getHeadNode()', () => {
+    const myLinkedList = new LinkedList()
+    myLinkedList.insert('first')
+    myLinkedList.insert('second')
+    myLinkedList.insert('third')
+
+    it('Returns an object', () => {
+      expect(myLinkedList.getHeadNode())
+        .to.be.a.object
+    })
+    it('Returns the first node in the list', () => {
+      expect(myLinkedList.getHeadNode().data)
+      .to.eql('first')
+    })
+  })
+  context('getTailNode()', () => {
+    const myLinkedList = new LinkedList()
+    myLinkedList.insert('first')
+    myLinkedList.insert('second')
+    myLinkedList.insert('third')
+
+    it('Returns an object', () => {
+      expect(myLinkedList.getTailNode())
+        .to.be.a.object
+    })
+    it('Returns the first node in the list', () => {
+      expect(myLinkedList.getTailNode().data)
+      .to.eql('third')
+    })
+  })
+  context('contains()', () => {
+    const myLinkedList = new LinkedList()
+    myLinkedList.insert('first')
+    myLinkedList.insert('second')
+
+    it('returns true the set contains the element', () => {
+      expect(myLinkedList.contains('first'))
+      .to.be.true
+    })
+    it('returns true if the element is last in the set', () => {
+      expect(myLinkedList.contains('second'))
+      .to.be.true
+    })
+    it('returns false if the set does not contain the element', () => {
+      expect(myLinkedList.contains('third'))
+      .to.be.false
+    })
+  })
+  context('find()', () => {
+    const myLinkedList = new LinkedList()
+    myLinkedList.insert('first')
+    myLinkedList.insert('second')
+
+    it('Returns the first node containing the provided data', () => {
+      expect(myLinkedList.find('second').data)
+      .to.eql('second')
+    })
+    it('Returns -1 if provided data is not found', () => {
+      expect(myLinkedList.find('third'))
+      .to.eql(-1)
+    })
+  })
+
+  context('insert()', () => {
+    it('Inserts a node (with data provided) to the tail', () => {
+      const myLinkedList = new LinkedList()
+
+      expect(() => myLinkedList.insert('foo'))
+        .to.alter(() => myLinkedList._size(), { from: 0, to: 1 })
+    })
+  })
+
+  context('insertFirst()', () => {
+    const myLinkedList = new LinkedList()
+    myLinkedList.insert('second')
+
+    it('Inserts a node to the head of the list', () => {
+      expect(() => myLinkedList.insertFirst('first'))
```

Comment:
personally, i like to only have one expect statement per it, the reason is so that errors are easier to track down EXACTLY where the problem is. If there are two expects per it, either expect could be failing and result in a failing of the entire statement.